cmd/federation-controller/main.go

```diff
@@ -210,56 +202,34 @@ func main() {
 		if err != nil {
 			log.Fatalf("failed to create FDS client: %v", err)
 		}
-		startFDSClient = func() {
-			go func() {
-				if err := federationClient.Run(); err != nil {
-					log.Errorf("failed to start FDS client, will reconnect in %s: %v", retryDelay, err)
-					time.AfterFunc(retryDelay, federationClient.Restart)
-				}
-			}()
-		}
 	}
 
-	if cfg.ConfigMode == config.ConfigModeMCP {
-		mcpServer := adss.NewServer(
-			&adss.ServerOpts{Port: 15010, ServerID: "mcp"},
-			meshConfigPushRequests,
-			startFDSClient,
-			mcp.NewGatewayResourceGenerator(istioConfigFactory),
-			mcp.NewServiceEntryGenerator(istioConfigFactory),
-			mcp.NewWorkloadEntryGenerator(istioConfigFactory),
-			mcp.NewVirtualServiceResourceGenerator(istioConfigFactory),
-			mcp.NewDestinationRuleResourceGenerator(istioConfigFactory),
-		)
-		if err := mcpServer.Run(ctx); err != nil {
-			log.Fatalf("Error running XDS server: %v", err)
-		}
-	} else if cfg.ConfigMode == config.ConfigModeK8s {
-		istioClient, err := istiokube.NewClient(istiokube.NewClientConfigForRestConfig(kubeConfig), "")
-		if err != nil {
-			log.Fatalf("failed to create Istio client: %v", err)
-		}
-
-		rm := kube.NewReconcilerManager(
-			meshConfigPushRequests,
-			kube.NewGatewayResourceReconciler(istioClient, istioConfigFactory),
-			kube.NewServiceEntryReconciler(istioClient, istioConfigFactory),
-			kube.NewWorkloadEntryReconciler(istioClient, istioConfigFactory),
-			kube.NewVirtualServiceReconciler(istioClient, istioConfigFactory),
-			kube.NewDestinationRuleReconciler(istioClient, istioConfigFactory),
-		)
-		if err := rm.ReconcileAll(ctx); err != nil {
-			log.Fatalf("initial Istio resource reconciliation failed: %v", err)
-		}
-		go rm.Start(ctx)
+	istioClient, err := istiokube.NewClient(istiokube.NewClientConfigForRestConfig(kubeConfig), "")
```

Comment:
Little nit, but I wonder if it makes sense to pass `ISTIO_META_CLUSTER_ID` here? If so - let's open an issue.

Reply:
I [requested](https://github.com/openshift-service-mesh/federation/pull/54#discussion_r1791844922) Eoin to leave it empty, because `CLUSTER_ID` is used by the client only in debug logs to distinguish clusters it connects to. We only connect to the local cluster, so I do not see the need to set the cluster id.

Reply:
`ISTIO_META_CLUSTER_ID` will only be set in the `istio-proxy` container, so this will end up in the empty string anyway. On the other hand, I do not like honoring additional environment variable in our container just to add the cluster name to a single debug message.

Reply:
If it's only about debug messaging then fair enough.